package.yaml

```
@@ -29,6 +28,25 @@ dependencies:
 library:
   source-dirs: src
 
+# Test suite
+tests:
+  test:
+    main: Main.hs
+    source-dirs:
+      - test
+    ghc-options:
+      - -threaded
+      - -rtsopts
+      - -with-rtsopts=-N
+      - -Wno-incomplete-uni-patterns # Failing at runtime just means failing the test, which is ok
+      - -O0
+    dependencies:
+      - HUnit
+      - hspec
+      - hspec-core
+      - hspec-expectations
+      - string-variants
```

Needs the `build-tools` stanza 
```suggestion
tests:
  test:
    main: Main.hs
    source-dirs:
      - test
    ghc-options:
      - -threaded
      - -rtsopts
      - -with-rtsopts=-N
      - -Wno-incomplete-uni-patterns # Failing at runtime just means failing the test, which is ok
      - -O0
    dependencies:
      - HUnit
      - hspec
      - hspec-core
      - hspec-expectations
      - string-variants
    build-tools:
      - hspec-discover
```

---

While this is outdated, it's still the right patch - `hspec-discover` needs to be under `build-tools` and not `dependencies`